src/classes/test/DataClient.js

```diff
@@ -0,0 +1,180 @@
+    member
+  } = mockData
+
+  t.is(await t.context.Client.memberCan(member, permission), true)
+})
+
+test.serial('permission level', async (t) => {
+  const {
+    member
+  } = mockData
+  t.is(await t.context.Client.permissionLevel(member), 0)
+})
+
+test.serial('combine tables', (t) => {
+  const {
+    defaultTables,
+    newTables
+  } = mockData
+  t.deepEqual(t.context.Client._combineTables(defaultTables, newTables), {
+    anime: 'Another',
+    shows: 'My little pony'
+  })
+})
+
+test.serial('get dirs', (t) => {
+  t.deepEqual(t.context.Client._getDirectories('src'), {
+    default: {
+      permissions: path.join(process.cwd(), `src/permissions/`),
+      commands: path.join(process.cwd(), `src/commands/`),
+      events: path.join(process.cwd(), `src/events/`),
+      settings: path.join(process.cwd(), `src/settings/`),
+      toggles: path.join(process.cwd(), `src/toggles/`)
+    },
+    user: {
+      permissions: path.join(process.cwd(), `src/permissions/`),
+      commands: path.join(process.cwd(), `src/commands/`),
+      events: path.join(process.cwd(), `src/events/`),
+      settings: path.join(process.cwd(), `src/settings/`),
+      toggles: path.join(process.cwd(), `src/toggles/`)
+    }
+  })
+})
+
+test.serial('load data', async (t) => {
+  const dir = path.join(process.cwd(), `src/permissions/`)
+  const files = await fs.readdir(dir)
+
+  t.context.Client._loadData('permissions', files, null, t.context.Client._permissionLoader)
+  t.true(t.context.loadData.calledOnce)
+})
+
+test.serial('load files', async (t) => {
+  const map = {
+    permissions: path.join(process.cwd(), `src/permissions/`),
+    commands: path.join(process.cwd(), `src/commands/`),
+    events: path.join(process.cwd(), `src/events/`),
+    settings: path.join(process.cwd(), `src/settings/`),
+    toggles: path.join(process.cwd(), `src/toggles/`)
+  }
+
+  const files = await fs.readdir(map.permissions)
+
+  t.context.Client._loadFiles(map, 'permissions', files, t.context.Client._permissionLoader)
+  t.true(t.context.log.called)
+})
+
+test.serial('load perm', (t) => {
+  t.context.Client._permissionLoader(permission)
+  t.true(t.context.permLoad.calledOnce)
+})
+
+test.serial('load command', (t) => {
+  const command = new Command(t.context.Client, {
+    name: 'test-command-name',
+    description: 'test-command-desc',
+    options: {
+      permission: 'Anyone'
+    },
+    run: function () {
+      return `Should return: ${this.name}`
+    }
+  })
+
+  t.context.Client._commandLoader(() => command)
+  t.true(t.context.cmdLoad.calledOnce)
+})
+
+test.serial('load event', (t) => {
+  t.context.Client._eventLoader(eventData)
+  t.true(t.context.eventLoad.calledOnce)
+})
+
+test.serial('load setting', (t) => {
+  t.context.Client._settingLoader(setting)
+  t.true(t.context.settingLoad.calledOnce)
+})
+
+test.serial('select loader', (t) => {
+  let perm = t.context.Client._permissionLoader
+  t.is(t.context.Client._selectLoader('permissions'), perm)
```

Comment:
actually a good test. nice one, add one for each loader function.

Reply:
do you mean that?

Reply:
*Pat*

Reply:
yep, but lets just make sure it's selecting the right loader in every case, and lets check what happens when there isn't a loader for something.